File: packages/python/src/unmsm_sum_api/schemas/course.py
from .common import BaseResponse, FlexibleModel


class Schedule(FlexibleModel):
    codHorario: int
    horaInicio: str
    horaFin: str
    dia: str


class Section(FlexibleModel):
    codSeccion: int
    codTurno: str
    canTopeAlumnos: int
    canAlumMatriculados: int
    codTipoHorAsignatura: str
    codDocente: str
    nomDocente: str
    apePaterno: str
    apeMaterno: str
    codAula: str
    horarios: list[Schedule]


class CourseScheduleItem(FlexibleModel):
    codAsignatura: str
    desAsignatura: str
    canHorasTeoria: int
    canHorasPractica: int
    canHorasLaboratorio: int
    numCreditaje: float
    indRegimen: str
    seccion: list[Section]


class CourseScheduleGroup(FlexibleModel):
    asignatura: list[CourseScheduleItem]


class CourseScheduleResponse(BaseResponse):
    data: list[CourseScheduleGroup]

File: packages/js/src/schemas/course.ts
import { z } from "zod";

import { BaseResponseSchema } from "./common.js";

export const ScheduleSchema = z.object({
  codHorario: z.number(),
  horaInicio: z.string(),
  horaFin: z.string(),
  dia: z.string(),
}).passthrough();

export const SectionSchema = z.object({
  codSeccion: z.number(),
  codTurno: z.string(),
  canTopeAlumnos: z.number(),
  canAlumMatriculados: z.number(),
  codTipoHorAsignatura: z.string(),
  codDocente: z.string(),
  nomDocente: z.string(),
  apePaterno: z.string(),
  apeMaterno: z.string(),
  codAula: z.string(),
  horarios: z.array(ScheduleSchema),
}).passthrough();

export const CourseScheduleResponseSchema = BaseResponseSchema.extend({
  data: z.array(z.object({
    asignatura: z.array(z.object({
      codAsignatura: z.string(),
      desAsignatura: z.string(),
      canHorasTeoria: z.number(),
      canHorasPractica: z.number(),
      canHorasLaboratorio: z.number(),
      numCreditaje: z.number(),
      indRegimen: z.string(),
      seccion: z.array(SectionSchema),
    }).passthrough()),
  }).passthrough()),
}).passthrough();

export type CourseScheduleResponse = z.infer<typeof CourseScheduleResponseSchema>;
